### PR TITLE
Closes #6289: Error Code when RUCSS can't be reached

### LINKED ISSUE #6289: Error code when RUCSS API can't be reached
**Context**
Related to this GH issue: https://github.com/wp-media/wp-rocket/issues/6222
QA team notices that part of the specification was missing in [this thread](https://wp-media.slack.com/archives/CUT7FLHF1/p1700747827387389): if the API can't be reached (not responding, timeout, ...) and that we don't have an HTTP answer, it is not clear how the plugin behaves. For instance, see [this possible issue](https://wp-media.slack.com/archives/CUT7FLHF1/p1700752455087559).
This means that we don't know exactly what happens in case the API can't be reached and we would not be able to identify the issue when debugging a user for instance.

**Expected behavior**
As mentioned in this [GH issue](https://github.com/wp-media/wp-rocket/issues/6222), we would need a specific error_code and/or error_message to be stored into the RUCSS table in the DB, in case the API can't be reached.

**Acceptance Criteria**
- If the API can be reached (normal case) there is no change of behavior (see https://github.com/wp-media/wp-rocket/issues/6222)
- Have a pending entry in RUCSS table. Disconnect from the internet or prevent the "fetch result" call to reach the API. Once the fetch has been tried, check the RUCSS table. The specific error_code and/or message should be registered for this job.

### PR DESCRIPTION
**Pull request by jeawhanlee**
## Description

This PR adds a default message when API response returns empty.

Fixes #6289

## Type of change

- Enhancement (non-breaking change which improves an existing functionality).

## Is the solution different from the one proposed during the grooming?

No

# Checklists

## Generic development checklist

- [x] My code follows the style guidelines of this project, with adapted comments and without new warnings.
- [x] The CI passes locally with my changes (including unit tests, integration tests, linter).

## Test summary

- [ ] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [ ] I validated all Acceptance Criteria of the related issues. (If applicable, provide proof).
- [ ] I validated all test plan the QA Review asked me to.

*If not, detail what you could not test.*

*Please describe any additional tests you performed.*

### CODE REVIEW

**Review by Miraeld — APPROVED** [review]

File: inc/Engine/Optimization/RUCSS/AbstractAPIClient.php [hunk 1]
```diff
@@ -125,6 +125,12 @@ private function check_response( $response ): bool {
 		if ( 200 !== $this->response_code ) {
 			$previous_errors = (int) get_transient( 'wp_rocket_rucss_errors_count' );
 			set_transient( 'wp_rocket_rucss_errors_count', $previous_errors + 1, 5 * MINUTE_IN_SECONDS );
+
+			if ( empty( $response ) ) {
+				$this->error_message = 'API Client Error';
```
Comment (wordpressfan) [inline inc/Engine/Optimization/RUCSS/AbstractAPIClient.php:130]: It's not a big deal but may I know why this string is not translatable?

Comment (CrochetFeve0251) [reply inc/Engine/Optimization/RUCSS/AbstractAPIClient.php:130]: @wordpressfan this string will be only visible inside the DB

**Review by wordpressfan — APPROVED** [review]

**Comment by Mai-Saad** [comment]
Related test plan https://wpmediaqa.testrail.io/index.php?/runs/view/796&group_by=cases:section_id&group_order=asc

**Comment by wordpressfan** [comment]
Good to go with the following note:
Now SaaS returns error 500 with navigation timeout so it's failing without any retries.

[testrail-report-554.pdf](https://github.com/wp-media/wp-rocket/files/13778466/testrail-report-554.pdf)

**Pull request merged**